### PR TITLE
Allow system conf modifiable and some take effect directly #1263

### LINKED ISSUE #1263: Dynamically modify some configuration at run time and restart to make effort for all configuration
**Is your feature request related to a problem? Please describe.**
Dynamically modify some configuration at run time and restart to make effort for all configuration

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

### PR DESCRIPTION
**Pull request by BossZou**
**What type of PR is this?**

api-change / bug / design / documentation / feature

**What this PR does / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #1263 

**Special notes for your reviewer:**

**Does this PR introduce a user-facing change?:**

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

File: core/src/scheduler/optimizer/BuildIndexPass.h [hunk 1]
```diff
@@ -39,6 +41,8 @@ class BuildIndexPass : public Pass {
     Run(const TaskPtr& task) override;
 
  private:
+    bool gpu_enable_;
```
Comment (JinHai-CN) [inline core/src/scheduler/optimizer/BuildIndexPass.h]: default value?

Comment (JinHai-CN) [inline core/src/scheduler/optimizer/BuildIndexPass.h]: default value?

File: core/src/scheduler/optimizer/FaissFlatPass.h [hunk 2]
```diff
@@ -40,6 +42,8 @@ class FaissFlatPass : public Pass {
     Run(const TaskPtr& task) override;
 
  private:
+    bool gpu_enable_;
```
Comment (JinHai-CN) [inline core/src/scheduler/optimizer/FaissFlatPass.h]: default value?

Comment (JinHai-CN) [inline core/src/scheduler/optimizer/FaissFlatPass.h]: default value?

File: core/src/scheduler/optimizer/FaissIVFFlatPass.h [hunk 3]
```diff
@@ -40,6 +42,8 @@ class FaissIVFFlatPass : public Pass {
     Run(const TaskPtr& task) override;
 
  private:
+    bool gpu_enable_;
```
Comment (JinHai-CN) [inline core/src/scheduler/optimizer/FaissIVFFlatPass.h]: default value?

Comment (BossZou) [reply core/src/scheduler/optimizer/FaissIVFFlatPass.h]: Resolved

File: core/src/scheduler/optimizer/FaissFlatPass.h (hunk 2 above)
Comment (BossZou) [reply core/src/scheduler/optimizer/FaissFlatPass.h]: Add default value : gpu_enable_=true

Comment (BossZou) [reply core/src/scheduler/optimizer/FaissFlatPass.h]: Add default value : gpu_enable_=true

File: core/src/scheduler/optimizer/BuildIndexPass.h (hunk 1 above)
Comment (BossZou) [reply core/src/scheduler/optimizer/BuildIndexPass.h]: Add default value : gpu_enable_=true

Comment (BossZou) [reply core/src/scheduler/optimizer/BuildIndexPass.h]: Add default value : gpu_enable_=true

**Review by fishpenguin — APPROVED** [review]

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**